R/RcodeBlock.R

```diff
@@ -0,0 +1,87 @@
+#' @title `RcodeBlock`
+#' @keywords internal
+RcodeBlock <- R6::R6Class( # nolint: object_name_linter.
+  classname = "RcodeBlock",
+  inherit = ContentBlock,
+  public = list(
+    #' @description Returns a `RcodeBlock` object.
+    #'
+    #' @details Returns a `RcodeBlock` object with no content and no parameters.
+    #'
+    #' @param content (`character(1)` or `character(0)`) a string assigned to this `RcodeBlock`
+    #' @param ... any `rmarkdown` R chunk parameter and it value.
+    #'
+    #' @return `RcodeBlock`
+    #' @examples
+    #' block <- teal.reporter:::RcodeBlock$new()
+    #'
+    initialize = function(content = character(0), ...) {
+      super$set_content(content)
+      self$set_params(list(...))
```

Comment:
I wonder if params should be moved up to ContentBlock as for example TableBlock could make use of params
https://github.com/insightsengineering/teal.reporter/blob/e3adbba61688df35e2ebca92efb9cd72956d16a6/R/Renderer.R#L157

Also as for now we're using the global knitr options, do we need to add the complexity of params for the blocks at this time?

Reply:
Good Question. It is strong suggestion of a small refactor here.
Then we should put the RcodeBlock before Picture and Table ones.
We should not put the params to ContentBlock as they are meaningless for TextBlock.
But the the moment the pictures ale loaded with regular markdown in the renderer, we not introduce the r code chunk with knitr picture load yet.

                        -> TextBlock
ContentBlock -> RcodeBlock  -> PictrueBLock
                                                   -> TableBlock

Not obvious one we could talk about htat.